apps/storefront/tests/shared/service/quickOrder.test.ts

```diff
@@ -0,0 +1,405 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+
+import { getBcOrderedProducts, getOrderedProducts } from '@/shared/service/b2b/graphql/quickOrder';
+import B3Request from '@/shared/service/request/b3Fetch';
+
+vi.mock('@/shared/service/request/b3Fetch', () => ({
+  default: {
+    graphqlB2B: vi.fn(),
+  },
+}));
+
+const mockB3Request = vi.mocked(B3Request);
+
+describe('QuickOrder GraphQL Service', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  describe('getOrderedProducts', () => {
+    it('includes companyId in query when provided', async () => {
+      const mockResponse = {
+        orderedProducts: {
+          totalCount: 10,
+          edges: [
+            {
+              node: {
+                id: '1',
+                productName: 'Test Product',
+                variantSku: 'TEST-SKU',
+                productId: 123,
+                variantId: 456,
+              },
+            },
+          ],
+        },
+      };
+
+      mockB3Request.graphqlB2B.mockResolvedValue(mockResponse);
+
+      const params = {
+        q: 'test',
+        first: 10,
+        offset: 0,
+        beginDateAt: '2023-01-01',
+        endDateAt: '2023-12-31',
+        orderBy: '-lastOrderedAt',
+        companyId: 789,
+      };
+
+      await getOrderedProducts(params);
+
+      expect(mockB3Request.graphqlB2B).toHaveBeenCalledWith({
+        query: expect.stringContaining('companyId: 789'),
+      });
+
+      const calledQuery = mockB3Request.graphqlB2B.mock.calls[0][0].query;
+      expect(calledQuery).toContain('q: "test"');
+      expect(calledQuery).toContain('first: 10');
+      expect(calledQuery).toContain('offset: 0');
+      expect(calledQuery).toContain('beginDateAt: "2023-01-01"');
+      expect(calledQuery).toContain('endDateAt: "2023-12-31"');
+      expect(calledQuery).toContain('orderBy: "-lastOrderedAt"');
+      expect(calledQuery).toContain('companyId: 789');
```

Comment:
you can now use `vitest-when` (similar to `jest-when`) to validate this:

```ts
when(mockB3Request.graphqlB2B)
  .calledWith(stringContainingAll('q: "test"', 'first: 10', 'offset: 0', ...))
  .thenReturn(mockResponse);

const response = await getOrderedProducts(params);

expect(response).toEqual(mockResponse);
```

or something on those lines.